### PR TITLE
Ipv4Block Read instance & parser

### PR DESCRIPTION
**Pull request by dsturnbull**
```
λ> AP.parseWith (return mempty) (I.whitespace *> I.ipv4Block) ("1.2.3.4/8")
Done "" (16909060,8)
```

```
λ> read "1.2.3.4/8" :: Ipv4Block
1.2.3.4/8
it :: Ipv4Block
```

### CODE REVIEW

File: src/HaskellWorks/Data/Network/Ip/Internal.hs [hunk 1]
```diff
@@ -45,3 +45,12 @@ ipv4Address = fourOctetsToWord32
 
 whitespace :: AP.Parser ()
 whitespace = void $ many (AP.satisfy isSpace)
+
+ipv4Mask :: AP.Parser Word8
```
Comment (jianwan2016) [inline src/HaskellWorks/Data/Network/Ip/Internal.hs]: nit: `ipv4NetMask`?

**Comment by jianwan2016** [comment]
👍  Probably a property test?

File: src/HaskellWorks/Data/Network/Ip/Internal.hs [hunk 2]
```diff
@@ -45,3 +45,12 @@ ipv4Address = fourOctetsToWord32
...
+ipv4Mask = octet
```
Comment (jianwan2016) [inline src/HaskellWorks/Data/Network/Ip/Internal.hs]: Oh, a mask is in the range of `1 - 32`?

File: src/HaskellWorks/Data/Network/Ip/Internal.hs [hunk 3]
```diff
@@ -45,3 +45,12 @@ ipv4Address = fourOctetsToWord32
...
+
+ipv4Block :: AP.Parser (Word32, Word8)
+ipv4Block = do
+  addr <- ipv4Address <* AP.char '/'
+  mask <- ipv4Mask
```
Comment (newhoggy) [inline src/HaskellWorks/Data/Network/Ip/Internal.hs]: Can merge to unblock, but we may want to exclude `33-255`

**Comment by newhoggy** [comment]
👍

**Comment by newhoggy** [comment]
👍

**Comment by jianwan2016** [comment]
👍

**Pull request merged**